src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -741,6 +755,40 @@ private InputsAndSuccessFlag selectInputs(TypedOperation operation) {
         continue;
       }
 
+      // If the user enables constant-tf-idf and we are determining a parameter for a class
+      // operation we will use a constant value extracted by constant-tf-idf under some probability.
+      if (GenInputsAbstract.constant_tfidf
+          && (operation instanceof TypedClassOperation && !isReceiver)
+          && Randomness.weightedCoinFlip(GenInputsAbstract.constant_tfidf_probability)) {
+        Log.logPrintf("Using constant from tf-idf as input.");
+        // Determine the scope for constant selection.
+        // It may be null, for the unnamed package (if the literal level is package).
+        @KeyFor("componentManager.scopeToConstantStatistics.scopeStatisticsMap") Object scopeKey =
+            componentManager.scopeToConstantStatistics.getScope(
+                ((TypedClassOperation) operation).getDeclaringType());
+
+        // Construct a list of candidate sequences that create values of type inputTypes[i].
+        Type neededType = operation.getInputTypes().get(i);
+        SIList<Sequence> candidates =
+            componentManager.getConstantSequences(neededType, isReceiver, scopeKey);
+
+        ConstantStatistics stats =
```

Comment:
Can you write a brief comment about the relationship between `candidates` and the keyset of `stats`?  In particular, if they differ, how are they different and what causes that?